sympy/stats/crv.py

```diff
@@ -531,11 +531,16 @@ def compute_density(self, expr, **kwargs):
 
         gs = solveset(expr - y, self.value, S.Reals)
 
-        if isinstance(gs, Intersection) and S.Reals in gs.args:
-            gs = list(gs.args[1])
-
-        if not gs:
-            raise ValueError("Can not solve %s for %s"%(expr, self.value))
+        if isinstance(gs, Intersection):
+            if len(gs.args) == 2:
+                if gs.args[0] is S.Reals:
+                    gs = gs.args[1]
+                else:
+                    raise NotImplementedError("Anticipating Intersection(Reals, Set)")
+            else:
+                raise NotImplementedError("Intersection length is not 2")
```

Comment:
What happens if NotImplementedError is raised here?

Does that go to the user or does it get caught somewhere?

Reply:
NotImplementedError will go to the user if the conditions are not met. I locally tried the following changes and it seems NotImplementedError is not getting caught anywhere.
![raise](https://github.com/sympy/sympy/assets/57264717/52c1b860-1c6e-4192-b693-d924f2bb4bab)

Reply:
I don't see the point of raising `NotImplementedError` here. The line below will already raise ValueError.

Reply:
Somewhat related questions were raised in (https://github.com/sympy/sympy/pull/25373#discussion_r1267618964)
[Final solution set returned by solveset](https://docs.sympy.org/latest/modules/solvers/solveset.html#what-are-the-general-methods-employed-by-solveset-to-solve-an-equation)
The above link confirms that the length of the arguments of intersection will always be 2, so I removed the corresponding raise.

Also, I observed S.Real if present in Intersection, it’s present at arg[0]. I tried to find code and a situation where it can be at args[1] (see pic), but I’m not sure how the order of S.Reals is changing within the implementation.
![Screenshot from 2024-01-20 03-13-00](https://github.com/sympy/sympy/assets/57264717/02980efb-32b7-4463-86e3-58fe3a901a70)